titiler/xarray/factory.py

```diff
@@ -228,40 +233,44 @@ def tiles_endpoint(  # type: ignore
         ) -> Response:
             """Create map tile from a dataset."""
             tms = self.supported_tms.get(tileMatrixSetId)
+            try:
+                with self.reader(
+                    url,
+                    variable=variable,
+                    group=z if multiscale else None,
+                    reference=reference,
+                    decode_times=decode_times,
+                    drop_dim=drop_dim,
+                    time_slice=time_slice,
+                    tms=tms,
+                    consolidated=consolidated,
+                ) as src_dst:
 
-            with self.reader(
-                url,
-                variable=variable,
-                group=z if multiscale else None,
-                reference=reference,
-                decode_times=decode_times,
-                drop_dim=drop_dim,
-                time_slice=time_slice,
-                tms=tms,
-                consolidated=consolidated,
-            ) as src_dst:
+                    image = src_dst.tile(
+                        x, y, z, tilesize=scale * 256, nodata=src_dst.input.rio.nodata
+                    )
 
-                image = src_dst.tile(
-                    x, y, z, tilesize=scale * 256, nodata=src_dst.input.rio.nodata
-                )
+                if post_process:
+                    image = post_process(image)
 
-            if post_process:
-                image = post_process(image)
+                if rescale:
+                    image.rescale(rescale)
 
-            if rescale:
-                image.rescale(rescale)
+                if color_formula:
+                    image.apply_color_formula(color_formula)
 
-            if color_formula:
-                image.apply_color_formula(color_formula)
-
-            content, media_type = render_image(
-                image,
-                output_format=format,
-                colormap=colormap,
-                **render_params,
-            )
+                content, media_type = render_image(
+                    image,
+                    output_format=format,
+                    colormap=colormap,
+                    **render_params,
+                )
 
-            return Response(content, media_type=media_type)
+                return Response(content, media_type=media_type)
+            except zarr.errors.GroupNotFoundError as e:
+                raise HTTPException(
```

Comment:
instead of a try/except we can add this error in the error handler 

in https://github.com/developmentseed/titiler-xarray/blob/9ac1686612d706e0f078a418818b16544efb11c0/titiler/xarray/main.py#L51-L53


```python
import zarr
from starlette import status

error_codes = {
    zarr.errors.GroupNotFoundError: status.HTTP_422_UNPROCESSABLE_ENTITY,
}

add_exception_handlers(app, error_codes)
```

Reply:
Thanks @vincentsarago that is much better, modified in https://github.com/developmentseed/titiler-xarray/pull/43/commits/b011743385efdc1de6d48307d043f60c68fffa9a